static/js/compose.js

```diff
@@ -389,15 +389,27 @@ exports.schedule_message = function schedule_message(request) {
     } else {
         request.to = JSON.stringify([request.to]);
     }
-    function success(data) {
-        notifications.notify_above_composebox('Scheduled your Message to be delivered at: ' + data.deliver_at);
-        $("#compose-textarea").attr('disabled', false);
-        clear_compose_box();
-    }
 
-    function error(response) {
-        $("#compose-textarea").attr('disabled', false);
-        compose_error(response, $('#compose-textarea'));
+    /* success and error callbacks are kind of a package deal here. When scheduling
+    a message either by means of slash command or from message feed, if we need to do
+    something special on success then we will also need to know if our request errored
+    and do something appropriate. Therefore we just check if success callback is not
+    defined and just assume request to be coming from compose box. This is correct
+    because we won't ever actually have success operate in different context than error. */
+    if (success === undefined) {
+        success = function (data) {
+            notifications.notify_above_composebox('Scheduled your Message to be delivered at: ' + data.deliver_at);
```

Comment:
I think a better pattern here is to just always have the called pass in these callbacks, so you avoid the funny conditional and long comment.  This will be especially nice if we split out a `reminder.js` that does the common code for popovers.js and compose.js.